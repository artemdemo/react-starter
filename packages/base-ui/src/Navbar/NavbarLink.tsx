import React from 'react';
import { NavLink } from 'react-router-dom';
import classNames from 'classnames';
import { TransparentButton } from '../TransparentButton/TransparentButton';

type TProps = {
  className?: string;
  to?: string;
  onClick?: (e?: any) => void;
  dataTestid?: string;
  children?: React.ReactNode;
};

export const NavbarLink: React.FC<TProps> = (props) => {
  const className = classNames('nav-link', props.className);

  const renderChildren = () => {
    if (props.to) {
      return (
        <NavLink
          className={className}
          to={props.to}
          onClick={props.onClick}
          end
          data-testid={props.dataTestid}
        >
          {props.children}
        </NavLink>
      );
    }

    return (
      <TransparentButton
        className={className}
        onClick={props.onClick}
        data-testid={props.dataTestid}
      >
        {props.children}
      </TransparentButton>
    );
  };

  return <span className="nav-item">{renderChildren()}</span>;
};
