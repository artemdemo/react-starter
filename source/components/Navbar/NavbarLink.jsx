import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { Link } from 'react-router-dom';

const NavbarLink = (props) => {
    return (
        <li
            className={classnames({
                'nav-item': true,
                'active': props.active,
            })}
        >
            <Link
                className='nav-link'
                to={props.to}
            >
                {props.children}
            </Link>
        </li>
    );
};

NavbarLink.propTypes = {
    to: PropTypes.string.isRequired,
    active: PropTypes.bool,
};

NavbarLink.defaultProps = {
    active: false,
};

export default NavbarLink;
