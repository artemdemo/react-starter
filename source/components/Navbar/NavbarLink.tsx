import React from "react";
import { NavLink } from "react-router-dom";
import styled from "styled-components";
import TransparentButton from "../TransparentButton/TransparentButton";

const NavItem = styled.div`
    display: inline-block;
`;

const ButtonItem = styled(TransparentButton)`
    padding: .5rem 1rem;
    line-height: inherit;
`;

type TProps = {
    to: string;
    exact: boolean;
    children?: any;
    onClick: (e?: any) => void;
}

type TState = {}

class NavbarLink extends React.PureComponent<TProps, TState> {
    static defaultProps = {
        to: undefined,
        onClick: undefined,
        exact: false,
    };

    renderChildren() {
        if (this.props.to) {
            return (
                // @ts-ignore
                <NavLink
                    className='nav-link'
                    to={this.props.to}
                    exact={this.props.exact}
                >
                    {this.props.children}
                </NavLink>
            );
        }
        return (
            <ButtonItem
                className='nav-link'
                onClick={this.props.onClick}
            >
                {this.props.children}
            </ButtonItem>
        );
    }

    render() {
        return (
            <NavItem>
                {this.renderChildren()}
            </NavItem>
        );
    }
}

export default NavbarLink;
