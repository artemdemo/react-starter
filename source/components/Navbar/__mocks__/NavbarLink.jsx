import React from 'react';
import _omit from 'lodash/omit';

const NavbarLink = props => (
    <div data-mock='NavbarLink'>
        {JSON.stringify(_omit(props, 'children'), null, 2)}
        {props.children}
    </div>
);

export default NavbarLink;
