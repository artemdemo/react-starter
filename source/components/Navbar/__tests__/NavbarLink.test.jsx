import React from 'react';
import renderer from 'react-test-renderer';
import NavbarLink from '../NavbarLink';

describe('NavbarLink', () => {
    it('render default', () => {
        const tree = renderer.create(
            <NavbarLink to='/test-link' />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render active', () => {
        const tree = renderer.create(
            <NavbarLink to='/test-link' active />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
