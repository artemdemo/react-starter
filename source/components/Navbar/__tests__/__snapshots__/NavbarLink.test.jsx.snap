// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarLink render default 1`] = `
<div
  className="sc-AxirZ jjcDso"
>
  <div
    data-mock="Link"
  >
    {
  "to": "/test-link"
}
  </div>
</div>
`;

exports[`NavbarLink should render active 1`] = `
<div
  className="sc-AxirZ jjcDso active"
>
  <div
    data-mock="Link"
  >
    {
  "to": "/test-link"
}
  </div>
</div>
`;
